### ISSUE TITLE
ICMP wrong checksum

### ISSUE THREAD

**Issue by david-cermak** [intel/net-test-suites#5]
Hi, 
I am having trouble with checksum of ICMP packets. 

It's actually the first packet sent as a part of `test_ct_init()` (do_ping). The icmp checksum of that packet is invalid (anyone else noticed?). I placed a workaround here https://github.com/intel/net-test-suites/commit/b4ccec4a2a12c5d85fc0f73f2f52ca5c4960186d.

**Comment by ozhuraki** [comment]
Hi David,

Thanks for your input.

I tested against an implementation that didn’t validate the ICMPv4 checksum and this slipped unnoticed.

I have made a quick fix, let me to know if it works for you.

I plan to refactor inet6_chksum() in port_data.cc and, perhaps, to add a unit or smoke test for it.

In case you notice other problems or have any enhancements/ideas, make an issue/pull request, or just drop me a mail.

Regards,
Oleg
